js/private/js_helpers.bzl

```diff
@@ -138,6 +138,43 @@ def gather_npm_package_store_deps(targets):
 
     return depset([], transitive = npm_package_store_deps)
 
+def copy_js_file_to_bin_action(ctx, file):
+    if ctx.label.workspace_name != file.owner.workspace_name or ctx.label.package != file.owner.package:
+        msg = """
+
+Expected to find source file {file_basename} in {this_package}, but instead it is in {file_package}.
+
+All source and data files that are not in the Bazel output tree must be in the same package as the
+target so that they can be copied to the output tree in an action.
+
+See https://github.com/aspect-build/rules_js/tree/dbb5af0d2a9a2bb50e4cf4a96dbc582b27567155/docs#javascript
```

Comment:
nit: link to docsite instead of github: https://docs.aspect.build/rules/aspect_rules_js/docs/#javascript